### PR TITLE
Add methods

### PR DESCRIPTION
**Pull request by patrickweh**
## Added
add event 'data-table-record-selected' when changing the state of the select checkbox of a row

add attributes to select

add attributes to table wrapper

## Fixed
SearchController.php now just creates a query builder instance if now search was given

### CODE REVIEW

File: src/DataTable.php [hunk 1]
```diff
@@ -379,9 +387,24 @@ public function loadData(): void
...
-        $baseQuery = $this->buildSearch();
-        $query = $baseQuery->clone();
+        $query = $this->buildSearch();
+        $baseQuery = $query->clone();
+
+        $result = $this->getResultFromQuery($query);
+
+        if ($aggregates = $this->getAggregate($baseQuery)) {
+            $result['aggregates'] = ! $this->search ? $aggregates : [];
+        }
```
Comment (SirSplasch) [inline src/DataTable.php:397]: New Line

File: src/DataTable.php [hunk 2]
```diff
@@ -379,9 +387,24 @@ public function loadData(): void
...
 
-        $baseQuery = $this->buildSearch();
-        $query = $baseQuery->clone();
+        $query = $this->buildSearch();
+        $baseQuery = $query->clone();
+
+        $result = $this->getResultFromQuery($query);
+
+        if ($aggregates = $this->getAggregate($baseQuery)) {
+            $result['aggregates'] = ! $this->search ? $aggregates : [];
```
Comment (SirSplasch) [inline src/DataTable.php:396]: What if $result is not an array?

File: src/DataTable.php [hunk 3]
```diff
@@ -401,12 +424,11 @@ public function loadData(): void
         } catch (QueryException $e) {
             $this->notification()->error($e->getMessage());
 
-            return;
+            return [];
```
Comment (SirSplasch) [inline src/DataTable.php:427]: return empty collection? So that return value cannot be array anymore

File: src/DataTable.php [hunk 4]
```diff
@@ -379,9 +387,24 @@ public function loadData(): void
...
-        $query = $baseQuery->clone();
+        $query = $this->buildSearch();
+        $baseQuery = $query->clone();
+
+        $result = $this->getResultFromQuery($query);
+
+        if ($aggregates = $this->getAggregate($baseQuery)) {
+            $result['aggregates'] = ! $this->search ? $aggregates : [];
+        }
+        $this->setData(is_array($result) ? $result : $result->toArray());
```
Comment (SirSplasch) [inline src/DataTable.php:398]: $result->toArray if $result is instance of LengthAwarePaginator intented?

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/DataTable.php (hunk 1 above)
Comment (patrickweh) [reply src/DataTable.php:397]: √

File: src/DataTable.php (hunk 2 above)
Comment (patrickweh) [reply src/DataTable.php:396]: moved below setData and set to $this->data

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**